Tasks/IISWebAppManagementOnMachineGroup/Utility.ps1

```diff
@@ -230,41 +246,84 @@ function Repair-Inputs([ref]$siteName, [ref]$physicalPath, [ref]$poolName, [ref]
         }
 
-        $sslCertThumbprint.Value = [Regex]::Replace($sslCertThumbprint.Value, "[^a-fA-F0-9]+" , "")
-
+        $sslCertThumbPrint.Value = [Regex]::Replace($sslCertThumbPrint.Value, "[^a-fA-F0-9]+" , "")
+        
         # Mark the SSL thumbprint value to be a secret value 
-        $sslCertThumbprintValue = $sslCertThumbprint.Value
-        Write-Host "##vso[task.setvariable variable=f13679253bf44b74afbd244ae83ca735;isSecret=true]$sslCertThumbprintValue"
+        if(($sslCertThumbPrint.Value.Length -ne 40) -or (-not [regex]::IsMatch($sslCertThumbPrint.Value, "[a-fA-F0-9]{40}"))){
+            throw (Get-VstsLocString -Key "InvalidSslThumbprintInBinding" -ArgumentList $protocol, $ipAddress, $port)
+        }
+
+        $sslCertThumbprintSecretValue = $sslCertThumbPrint.Value
+        Write-Host "##vso[task.setvariable variable=f13679253bf44b74afbd244ae83ca735;isSecret=true]$sslCertThumbprintSecretValue"
     }
 }
 
 function Test-Inputs
 {
     param (
-        [string] $virtualPath,
-        [string] $sslCertThumbPrint,
-        [string] $protocol, 
-        [string] $addBinding
+        [string] $virtualPath
     )
 
-    if((-not [string]::IsNullOrWhiteSpace($sslCertThumbPrint)) -and ($protocol -ieq "https") -and ($addBinding -ieq "true")) 
+    if((-not [string]::IsNullOrWhiteSpace($virtualPath)) -and (-not $virtualPath.StartsWith("/")))
     {
-        if(($sslCertThumbPrint.Length -ne 40) -or (-not [regex]::IsMatch($sslCertThumbPrint, "[a-fA-F0-9]{40}")))
-        {
-            throw (Get-VstsLocString -Key "InvalidSslThumbprint" )
-        }
+        throw (Get-VstsLocString -Key "InvalidVirtualPath")
     }
+}
 
-    if((-not [string]::IsNullOrWhiteSpace($virtualPath)) -and (-not $virtualPath.StartsWith("/")))
+function Get-HostName
+{
+    param(
+        [string]$protocol,
+        [string]$hostNameWithHttp,
+        [string]$hostNameWithSNI,
+        [string]$hostNameWithOutSNI,
+        [string]$sni
+    )
+    $hostName = [string]::Empty
+
+    if($protocol -eq "http")
     {
-        throw (Get-VstsLocString -Key "InvalidVirtualPath")
+        $hostName = $hostNameWithHttp
+    }
+    elseif($sni -eq "true")
+    {
+        $hostName = $hostNameWithSNI
+    }
+    else
+    {
+        $hostName = $hostNameWithOutSNI
+    }
+    return $hostName
+}
+
+function Validate-Bindings {
+    param (
+        [string] $bindings
+    )
+    
+    $bindingsObj = $bindings | ConvertFrom-Json 
+
+    foreach ($binding in $bindingsObj.bindings) {
+
+        if($binding.protocol -eq "https") {
+            if(-not [string]::IsNullOrWhiteSpace($binding.sslThumbprint)) {
+                # Trim all non-hexadecimal characters from the ssl cetificate thumbprint
+                $sslCertThumbPrint = $binding.sslThumbPrint
+                $sslCertThumbPrint = Repair-Inputs -sslCertThumbPrint ([ref]$sslCertThumbPrint) -ipAddress $binding.ipAddress -protocol $binding.protocol -port $binding.port
```

Comment:
except cert thumbprint, other inputs are not really inputs for repairing. The Name of the method and signature is not matching.